remote.go

```diff
@@ -1387,7 +1387,7 @@ func pushHashes(
 	allDelete bool,
 ) (*packp.ReportStatus, error) {
 
-	rd, wr := ioutil.Pipe()
+	rd, wr := io.Pipe()
```

Comment:
The existing pipe implementation seems to differ from the std library by having non blocking writes and for blocking reads until the writer is closed. How can we ensure this won't cause regressions? I think some of the errors we are getting may be related to that:

```
remote_test.go:1484:
    c.Assert(err, IsNil)
... value *fs.PathError = &fs.PathError{Op:"write", Path:"|1", Err:0x20} ("write |1: broken pipe")
```

Reply:
Not really, the [bufpipe](https://github.com/acomagu/bufpipe) library is only used for [`js` builds](https://github.com/go-git/go-git/blob/master/utils/ioutil/pipe_js.go), all other builds use the std library which doesn't make sense :/

I'm going to put this change in a separate PR